TESTS/icb_arpack_c.c

```diff
@@ -109,38 +107,36 @@ int zn() {
   a_int iparam[11];
   a_int ipntr[14];
   a_dcomplex workd[3 * N];
-  a_int rvec = 0;
+  a_int rvec = 0; // eigenvectors omitted
   char howmny[] = "A";
   a_dcomplex* d =
-      (a_dcomplex*)malloc((nev + 1) * sizeof(a_dcomplex));
-  a_int select[ncv];
+      (a_dcomplex*)malloc(nev * sizeof(a_dcomplex));
+  a_int select[ncv]; // since HOWMNY = 'A', only used as workspace here
   int i; // C99 compliant.
-  for (i = 0; i < ncv; i++) select[i] = 1;
-  a_dcomplex z[(N + 1) * (nev + 1)];
-  a_int ldz = N + 1;
-  a_dcomplex sigma = CMPLX(0., 0.);
+  a_dcomplex z[N * nev];
+  a_int ldz = N;
+  a_dcomplex sigma = CMPLX(0., 0.); // not referenced in this mode
   int k;
   for (k = 0; k < 3 * N; ++k) workd[k] = 0;
-  a_dcomplex workl[3 * (ncv * ncv) + 6 * ncv];
-  for (k = 0; k < 3 * (ncv * ncv) + 6 * ncv; ++k) workl[k] = 0;
-  a_int lworkl = 3 * (ncv * ncv) + 6 * ncv;
+  a_dcomplex workl[ncv * (3 * ncv + 5)];
+  for (k = 0; k < ncv * (3 * ncv + 5); ++k) workl[k] = 0;
+  a_int lworkl = ncv * (3 * ncv + 5);
   double rwork[ncv];
   a_dcomplex workev[2 * ncv];
   a_int info = 0;
 
-  iparam[0] = 1;
-  iparam[2] = 10 * N;
-  iparam[3] = 1;
-  iparam[4] = 0;  // number of ev found by arpack.
```

Comment:
Can you let iparam[4] initialized to 0?
Initialisation is always good thing even if it's an output.

Reply:
For the initialization discussion: I am neutral in choosing to initialize (or not) the output-only arguments, but inclined to have a consistent treatment of all these arguments in the same function. If we choose to initialize iparam[4] because "initialization is always a good thing", by the same token we should also initialize all other elements in "iparam[0:10]", and also all elements in "ipntr[:]", and also all elements in the eigenvalues -- the list simply goes on... So I feel like just not necessary to do so just particularly for iparam[4], since it is not special at all compared to other variables mentioned above.

What do you think? If you want, I can still switch on ipram[4] initialization...